### PR TITLE
Fixes the normals of SphereMesh when the radius is different from 1

### LINKED ISSUE #52965: Wrong SphereMesh shading in 3.4 beta 5
### Godot version

3.4 beta 5 

### System information

MacOS M1

### Issue description

The default shading seems to be broken. The shading in beta 5 looks as if the light comes from the top, even though the directional light comes from the upper left in the screenshots.

The issue does not occur in Beta 4

### 3.4 Beta 5
3 4 beta 5

### Steps to reproduce

Create a scene with a mesh and a directional light.

### Minimal reproduction project

[ExpeditionSim.zip](https://github.com/godotengine/godot/files/7216207/ExpeditionSim.zip)

### PR DESCRIPTION
**Pull request by drcd1**
Fixes https://github.com/godotengine/godot/issues/52965.

The issue was that there was a "typo" in the previous code. The wrong coordinates were being multiplied by the radius. I didn't see it last time because I only tested the mesh for radius == 1. 

I have tested it now for varying values of "radius" and "height" when creating the sphere mesh, and compared the results with applying a transform to the whole mesh, so everything should be working now.

*Bugsquad edit: Redone version of https://github.com/godotengine/godot/pull/51995.*

### CODE REVIEW

File: scene/resources/primitive_meshes.cpp [hunk 1]
```diff
@@ -1466,7 +1466,7 @@ void SphereMesh::_create_mesh_array(Array &p_arr) const {
 			} else {
 				Vector3 p = Vector3(x * radius * w, y, z * radius * w);
 				points.push_back(p);
-				Vector3 normal = Vector3(x * radius * w * scale, y / scale, z * radius * w * scale);
+				Vector3 normal = Vector3(x * w * scale, radius * (y / scale), z * w * scale);
 				normals.push_back(normal.normalized());
```
Comment (BastiaanOlij) [inline scene/resources/primitive_meshes.cpp:1470]: Probably a dumb question but why is it `* scale` on `x` and `z`, and `/ scale` on `y`?

Comment (drcd1) [reply scene/resources/primitive_meshes.cpp:1470]: scale is basically just the "vertical radius", so you can think of a the ellipsoid as a unit sphere scaled by "scale" in the y axis and by "radius" in the x and z axis.

Because of this, the normal actually needs to be scaled in the x and z axis by `scale`, and in the y axis by `radius`. This is a particular case of using the transpose of the inverse of the transform matrix, which one usually does when transforming normals. See my explanation in https://github.com/godotengine/godot/pull/51995.

The `/scale` factor on `y` comes from the fact that `y`is already scaled vertically  (since `y = scale * cos(Math_PI * v);`), so by dividing by scale we get `cos(Math_PI*v)`

To sum up:
If `(x,y,z)` is a point in the unit sphere, the (not normalized) normal of the scaled ellipsoid is given by:
`normal = (x*scale,y*radius,z*scale)`

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.4. (Together with #51995.)
